SmartTutor/ContentModel/ChallengeService.cs

```diff
@@ -27,6 +28,16 @@ public Challenge GetChallenge(int challengeId)
             return _learningObjectRepository.GetChallenge(challengeId);
         }
 
+        public List<ChallengeHint> GetAllHints(int challengeId)
+        {
+            return GetChallenge(challengeId).GetAllChallengeHints();
+        }
+
+        public List<ChallengeHint> GetHintsForSolutionAttempt(int challengeId, List<CaDETClass> solutionAttempt)
```

Comment:
As far as I can tell, this function supports the same Use case as the CheckChallengeCompletion function. We supply sourceCode and a challengeId and get an evaluation of our work. The only difference is that this does not include a bool for challenge completion. There is no need for this method.